sysconfig/cloudinit.go

```diff
@@ -743,6 +744,16 @@ func CloudInitStatus() (CloudInitState, error) {
 		return CloudInitDisabledPermanently, nil
 	}
 
+	// if it was explicitly disabled via the kernel commandline, then
+	// return special status for that, if commandline is unavailable,
+	// assume not disabled
+	cmdline, err := kcmdline.KeyValues("cloud-init")
+	if err != nil {
+		logger.Noticef("cannot obtain cloud-init from kernel command line: %v", err)
+	} else if err == nil && cmdline["cloud-init"] == "disabled" {
```

Comment:
sorry, should have caught this earlier:
```suggestion
	} else if cmdline["cloud-init"] == "disabled" {
```